src/Migration/History/LogGroupingService.php

```diff
@@ -0,0 +1,423 @@
+        ?string $filterCode,
+        ?string $filterStatus,
+        ?string $filterEntity,
+        ?string $filterField,
+        Context $context,
+    ): array {
+        $run = $this->getMigrationRunForLogs($runUuid, $context);
+        $connectionIdBytes = Uuid::fromHexToBytes($run->getConnectionId() ?? '');
+        $runIdBytes = Uuid::fromHexToBytes($runUuid);
+
+        $params = $this->buildParams(
+            $runIdBytes,
+            $level,
+            $page,
+            $limit,
+            $connectionIdBytes,
+            $filterCode,
+            $filterEntity,
+            $filterField
+        );
+
+        $whereClause = $this->buildWhereClause($filterCode, $filterEntity, $filterField);
+        $havingClause = $this->buildHavingClause($filterStatus);
+        $orderByClause = $this->buildOrderByClause($sortBy, $sortDirection);
+
+        $sql = $this->buildQuery(
+            $whereClause,
+            $havingClause,
+            $orderByClause,
+            $filterStatus !== null
+        );
+
+        $result = $this->connection->executeQuery($sql, $params, [
+            'limit' => ParameterType::INTEGER,
+            'offset' => ParameterType::INTEGER,
+        ]);
+
+        $rows = $result->fetchAllAssociative();
+        $total = \count($rows) > 0 ? (int) $rows[0]['total'] : 0;
+
+        $levelCounts = $this->getLogLevelCounts(
+            $params,
+            $whereClause,
+            $havingClause,
+            $filterStatus !== null
+        );
+
+        return [
+            'total' => $total,
+            'items' => $this->mapLogsFromRows($rows),
+            'levelCounts' => $levelCounts,
+        ];
+    }
+
+    /**
+     * @throws Exception
+     *
+     * @return array<string>
+     */
+    public function getAllLogIdsByCodeAndEntity(
+        string $runId,
+        string $code,
+        string $entityName,
+        string $fieldName,
+        ?string $connectionId = null,
+    ): array {
+        $join = '
+            LEFT JOIN swag_migration_fix f ON (
+                f.entity_name = l.entity_name
+                AND f.path = l.field_name
+                AND f.entity_id = l.entity_id
+        ';
+
+        $params = [
+            'runId' => Uuid::fromHexToBytes($runId),
+            'code' => $code,
+            'entityName' => $entityName,
+            'fieldName' => $fieldName,
+        ];
+
+        if ($connectionId !== null && $connectionId !== '') {
+            $join .= ' AND f.connection_id = :connectionId';
+            $params['connectionId'] = Uuid::fromHexToBytes($connectionId);
+        }
+
+        $join .= ')';
+
+        $sql = "
+            SELECT LOWER(HEX(l.id)) as id
+            FROM swag_migration_logging l
+            {$join}
+            WHERE l.run_id = :runId
+                AND l.code = :code
+                AND l.entity_name = :entityName
+                AND l.field_name = :fieldName
+                AND l.user_fixable = 1
+                AND f.id IS NULL
```

Comment:
Similar to the other endpoint, this is an unrestricted query, potentially returning lots of ids. We should benchmark this service and see if it actually becomes an issue